### PR TITLE
Legal cleanup

### LINKED ISSUE #1576: Legal cleanup
From @mlinksva:

>a few comments on the license notices available (as 'Acknowledgements') from the `GitHub Desktop | About GitHub Desktop` menu:
>
>- I see that some dependencies listed only show the name of the license and don't include a copy of the license. That's fine if the dependency doesn't provide a copy of the license itself, but the [first such dependency I checked](https://github.com/webpack/loader-utils) does provide a copy, which ought be included in the acknowledgements
>- It'd be nice to include the project's MIT license at the top, as the Terms and Conditions don't grant any other license for the app. Take a look at `Atom | View License`.
>- Along those lines, I suggest the 'linktext' to be 'License and Open Source Notices' rather than 'Acknowledgements' (also reflecting what the Terms and Conditions say to expect)
>- I can't select or copy text in the acknowledgements popup (unlike the T&C one). Just an annoyance. If it weren't just a bunch of friendly open source licenses though, it would really unfriendly behavior.

### PR DESCRIPTION
**Pull request by shiftkey**
Fixes #1576 

 - [x] use a more accurate label for the dialog and link
 - [x] display Desktop license from `legal-eagle` output
 - [x] make license text selectable wherever we render it
 - [x] update `legal-eagle` to get new package licenses
 - ~~[ ] collapse license text to make this list less crazy to scroll through~~ :boot:

### CODE REVIEW

**Comment by shiftkey** [comment]
Screenshot time:



The only thing I don't really like here is that it's less clear where we switch between the main license and the libraries licences:



I'll think about that while waiting on clarification about one of the tasks, but suggestions are welcome.

File: app/src/ui/acknowledgements/acknowledgements.tsx [hunk 1]
```diff
@@ -34,10 +34,21 @@ export class Acknowledgements extends React.Component<IAcknowledgementsProps, IA
   public constructor(props: IAcknowledgementsProps) {
     super(props)
 
-    this.state = { licenses: null }
+    this.state = { licenses: null, licenseText: null }
   }
 
   public componentDidMount() {
+    const licensePath = Path.join(getAppPath(), 'static', 'LICENSE')
```
Comment (joshaber) [inline app/src/ui/acknowledgements/acknowledgements.tsx]: Our license dump actually already includes our license, we explicitly skip it: https://github.com/desktop/desktop/pull/1611/files#diff-bbe735d95bc2a145bffe0c707c9b00a4L62

Comment (shiftkey) [inline app/src/ui/acknowledgements/acknowledgements.tsx]: Hell yes, I'll steal that and save the work.

Comment (shiftkey) [inline app/src/ui/acknowledgements/acknowledgements.tsx]: LOL, actually this suffers from the same problem as mentioned here https://github.com/desktop/desktop/issues/1576#issuecomment-302363884

Comment (joshaber) [inline app/src/ui/acknowledgements/acknowledgements.tsx]: lol-laugh-out-loud

**Comment by shiftkey** [comment]
I've opened https://github.com/desktop/desktop/issues/1846 to track some polish work we can do related to this view, but I won't hold this PR up further.

**Pull request merged**